BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/ShoppingCartWorkflowImpl.java

```diff
@@ -0,0 +1,98 @@
+package barryspeanuts;
+
+import barryspeanuts.model.Address;
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.PurchaseItem;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowQueue;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+public class ShoppingCartWorkflowImpl implements ShoppingCartWorkflow {
+  private static final Logger logger = Workflow.getLogger(ShoppingCartWorkflowImpl.class);
+
+  boolean exit = false;
+  List<PurchaseItem> purchaseItems = new ArrayList<>();
+  private final WorkflowQueue<Runnable> queue = Workflow.newWorkflowQueue(1024);
+
+  @Override
+  public void startWorkflow() {
+    logger.info("Starting Workflow for Barry's Peanuts");
+    Workflow.await(() -> exit);
+  }
+
+  @Override
+  public List<PurchaseItem> queryPurchaseItems() {
+    return this.purchaseItems;
+  }
+
+  @Override
+  public void addItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.add(purchaseItem);
+  }
```

Comment:
This can be removed, `addItems` is all you need (if you need to add a single item, you can still use that one). Makes no sense to have both.